src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/RemoteVettingController.php

```diff
@@ -189,35 +180,30 @@ public function remoteVetMatchAction(Request $request, $processId)
         try {
             $command->matches = $this->remoteVettingService->getAttributeMatchCollection($localAttributes);
 
-            // todo: add command assertions
-
             $form = $this->createForm(RemoteVetValidationType::class, $command)->handleRequest($request);
             if ($form->isSubmitted() && $form->isValid()) {
 
                 /** @var RemoteVetValidationCommand $command */$command = $form->getData();
 
-                $this->remoteVettingService->done(
+                $token =  $this->remoteVettingService->done(
                     ProcessId::create($processId),
                     $this->getIdentity(),
                     $localAttributes,
                     $command->matches,
                     (string)$command->remarks
                 );
 
-                $flashBag->add('success', 'ss.second_factor.revoke.alert.remote_vetting_successful');
-
-                // todo: vet token
-                //
-                //                $command = new RemoteVetCommand();
-                //                $command->identity = $user->getIdentityId();
-                //                $command->secondFactor = $user->getSecondFactorId();
-                //            /** @var SecondFactorService $service */
-                //        $service = $this->get('surfnet_stepup_self_service_self_service.service.second_factor');
-                //            if ($service->remoteVet($command)) {
-                //                $flashBag->add('success', 'ss.second_factor.revoke.alert.remote_vetting_successful');
-                //            } else {
-                //                $flashBag->add('error', 'ss.second_factor.revoke.alert.remote_vetting_failed');
-                //            }
+                $command = new RemoteVetCommand();
+                $command->identity = $token->getIdentityId();
+                $command->secondFactor = $token->getSecondFactorId();
```

Comment:
Do we not want a vetting mode included in the event?

Reply:
The vetting mode isn't necessary because there currently is a dedicated `RemoteVetSecondFactorCommand` so there is a distinction between RV and RA vetting. Maybe we want to add the name of the used RV IdP to the command. Add it to the Pivotal issue above.